src/dtgtk/thumbnail.c

```diff
@@ -1377,12 +1388,14 @@ void dt_thumbnail_resize(dt_thumbnail_t *thumb, int width, int height, gboolean
   thumb->width = width;
   thumb->height = height;
   gtk_widget_set_size_request(thumb->w_main, width, height);
+
   // file extension
-  gtk_widget_set_margin_start(thumb->w_ext, 0.045 * width);
+  gtk_widget_set_margin_start(thumb->w_ext, 0.03 * width);
+  gtk_widget_set_margin_top(thumb->w_ext, 0.03 * width);
 
   // retrieves the size of the main icons in the top panel, thumbtable overlays shall not exceed that
   int max_size = darktable.gui->icon_size;
-  const int fsize = fminf(max_size, .09 * width);
+  const int fsize = fminf(max_size, 0.94 * width / 11.0f);
 
```

Comment:
just out of curiosity : do you find all those values by try and guess or they comes for some clever computations ?

Reply:
The margins are set to 3% on each side, so we get 94% of the full width for the inner content. In this width:
```
    // we need to squeeze 5 stars + 1 reject + 1 colorlabels symbols on a thumbnail width
    // stars + reject having a width of 2 * r1 and spaced by r1 => 18 * r1
    // colorlabels => 3 * r1 + space r1
    // inner margins are 0.02 * width
```
so we have 22 × r1 to squeeze in 94% of the width, each icon having a diameter of 2 × r1, hence 11 diameters. But for some reason, the font size was not using the icon radius as a base height, so this only sync font size on actual icon diameter.